features/support/api_helpers.rb

```diff
@@ -364,7 +364,15 @@ def create_user(user_role, custom_user_data = {})
 
   response = call_api(:post, "/users", payload: { users: user_data, updated_by: 'functional_tests' })
   expect(response.code).to eq(201), response.body
+
   @user = JSON.parse(response.body)["users"]
+
+  if (custom_user_data['active'] || '').downcase == "false"
+    # this can't be specified at creation time so we need to follow up with an update to the user
+    response = call_api(:post, "/users/#{@user['id']}", payload: { users: {'active': false}, updated_by: 'functional_tests' })
+    expect(response.code).to eq(200), response.body
+    @user = JSON.parse(response.body)["users"]
+  end
```

Comment:
Could we specify an inactive user in the settings (`DM_INACTIVE_BUYER_USER_EMAIL`) so that we can avoid creating one each time, or randomly pick an inactive user somehow (we have more inactive suppliers than buyers, so maybe that would be a better bet)?

Reply:
:thinking: Well, yeah we could but really we already create a new user every time we go `And I have a buyer user` or `And I have a supplier user` currently. Our APIs don't exactly make this sort of "filtering for arbitrary properties" like active-ness particularly easy.